File: Array/Prob05/Prob05.js
const solution = (arr) => {
    let finderPointer = 0;
    let swapPointer = 0;

    for (let i = 0; i < arr.length; i++) {
        if (arr[i] >= 0) finderPointer++;
        else {
            for (let j = i; j > swapPointer; j--) {
                let temp = arr[j];
                arr[j] = arr[j - 1];
                arr[j - 1] = temp;
            }
            swapPointer++;
        }
    }

    arr.forEach(element => {
        console.log(element);
    });
};

arr = [ 12, 3, -2, 9, -5, -3, 7 ];
solution(arr);

File: Array/Prob05/Prob05.java
public class Prob05 {

    static int[] arr = { 12, 3, -2, 9, -5, -3, 7 };

    public static void swap(int index1, int index2) {
        int temp = arr[index1];
        arr[index1] = arr[index2];
        arr[index2] = temp;
    }
    public static void main(String[] args) {
        int len = arr.length;

        int swapP = 0, finderP = 0;

        for (int i = 0; i < len; i++) {
            if (arr[i] >= 0) finderP++;
            else {
                for (int j = i; j > swapP; j--)
                    swap(j, j-1);
                swapP++;
            }
        }

        for (int element : arr)
            System.out.println(element);
    }
}